### ISSUE TITLE
DMI_EMPTY_DB_PASSWORD and DMI_CONSTANT_DB_PASSWORD miss value from ternary operator

### ISSUE THREAD

**Issue by csrma** [spotbugs/spotbugs#3529]
When the blank string (`DMI_EMPTY_DB_PASSWORD`) or hard-coded string (`DMI_CONSTANT_DB_PASSWORD`) is passed from a ternary operator, neither of the rules are detected by Spotbugs. Below is an example code for this.
```java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

class BuggyCode {
    private static final String DATABASE_URL = "jdbc:mysql://localhost:3306/mydatabase";
    private static final String DATABASE_USER = "myuser";
    private boolean passwordDecider = false;

    public Connection showBug() throws SQLException {
        Connection conn = DriverManager.getConnection(DATABASE_URL, DATABASE_USER, passwordDecider ? "password" : "");    // line 11 - should raise error
        return conn;
    }
}
```
***Expected Outcome:*** Both hardcoded password and blank password error raised on line 11.

Spotbugs: `4.9.3`
Java: `Jdk 17` and `Jdk 21`
Run Method: `CLI` as:
```bash
javac BuggyCode.java && \
    java -jar $SPOTBUGS_JAR -textui -effort:max *.class
```
